documentation/platyps-help.md

```diff
@@ -0,0 +1,227 @@
+
+This command generates the MAML based solely on the markdown files. All the information from the cmdlets (thru reflection) are already incorporated into the markdown thru New-MarkdownHelp and Update-MarkdownHelp (or Update-MarkdownHelpModule). This means that you won't need to build/install/import modules to generate help for them.
+
+### Previewing MAML
+
+After you have created or updated your markdown for the cmdlets, you can preview what the corresponding MAML will look like in PowerShell with the `Get-HelpPreview` cmdlet. This cmdlet will generate the MAML from the markdown files and display it in PowerShell so you can see if the help looks correct.
+
+Running the following command will display the help for **all** of your cmdlets:
+```powershell
+Get-HelpPreview -Path $path
+```
+where $path is the path to the MAML help file you generated.
+
+To get the help for a single cmdlet, run the following command:
+```powershell
+$help = Get-HelpPreview -Path $path
+$help | Where-Object {  $_.Name -eq $cmdlet }
+```
+
+## Getting Started
+
+If your service does not currently have any markdown help, follow the below steps to get started.
+
+### Running the `GenerateHelp.ps1` script
+
+If you want to automate this process, run the `tools\platyPS\GenerateHelp.ps1` script with the following command:
+```powershell
+.\GenerateHelp.ps1 -Service $service -PathToRepo $pathToRepo
+```
+where `$service` is the name of your service (*e.g.,* Profile, Compute, Network, etc.) and `$pathToRepo` is the path to the **azure-powershell** repo locally on your machine.
+
+This script can be used for new services looking to create markdown files, or services looking to update their existing markdown files.
+
+### Steps for creating help
+
+**1: Create a folder for your service**
+
+In the `azure-powershell` repository, navigate to the `help` folder in the root, and, if your service does not already have one, create a new folder that will hold the markdown files for your service
+
+**2: Import your module**
+
+Run the following command in PowerShell:
+
+```powershell
+Import-Module "PATH_TO_REPO\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.SERVICE\AzureRM.SERVICE.psd1"
+```
+
+**3: Generate the markdown files**
+
+Run the following command in PowerShell:
+
+```powershell
+New-MarkdownHelp -Module AzureRM.SERVICE -OutputFolder "PATH_TO_REPO\help\SERVICE" -WithModulePage
+```
+
+This will generate a markdown file for each cmdlet in your MAML file. Make sure that all of the cmdlets that you have help for in the MAML file have a corresponding markdown file in the folder.
+
+**4: Generate the MAML file**
+
+Run the following command in PowerShell:
+
+```powershell
+New-ExternalHelp -Path "PATH_TO_REPO\help\SERVICE" -OutputPath "PATH_TO_REPO\src\ResourceManager\SERVICE\Commands.SERVICE -Force"
```

Comment:
This command must be run every time there is a change made to a markdown file or the cmdlets. The `GenerateHelp` script always runs this command, no matter if the user is on-boarding or has existing markdown.